ComfySoda_Unity2018.2.7f1/Assets/Scripts/ComySodaEngine/EnemyController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Unity;
+using UnityEditor;
+using UnityEngine;
+
+
+namespace cs
+{
+    public class EnemyController : MonoBehaviour
+    {
+
+        #region Static referencing
+        public static EnemyController enemyController = null;
+        private void Awake()
+        {
+            Debug.Assert(enemyController == null, "Error: There should only be one enemy controller assigned to \"EnemyController.enemyController\" at a time");
+            enemyController = this;
+        }
+
+        private void OnDestroy()
+        {
+            Debug.Assert(enemyController == this, "Error: The enemy controller assigned to \"EnemyController.enemyController\" does not equal: " + gameObject);
+            enemyController = null;
+        }
+        #endregion
+
+
+        #region Enemy distance
+        [SerializeField]
+        public float CLOSE = 7;
+        [SerializeField]
+        public float MID = 12;
+
+        public enum EnemyRange { CLOSE, MID, FAR};
+        public EnemyRange enemyRange = EnemyRange.FAR;
+        public List<EnemyLogic> enemyList = EnemyLogic.enemyList;
+
```

Comment:
These also are serialized. Given your naming convention, I don't think you want that. Otherwise your naming convention is nonexistent.

Reply:
serialization is undesirable here, but it needs to be referenced by other classes.